src/modules/to-do-manager.js

```diff
@@ -0,0 +1,144 @@
+class ToDoListManager {
+  dataFinder = 'to-do-list';
+
+  constructor() {
+    this.tasks = [];
+
+    this.queryTasks();
+  }
+
+  queryTasks() {
+    const tempList = localStorage.getItem(this.dataFinder);
+    if (tempList !== null) this.tasks = JSON.parse(tempList);
+  }
+
+  commitTasks() {
+    localStorage.setItem(this.dataFinder, JSON.stringify(this.tasks));
+  }
+
+  static get listView() {
+    return document.body.querySelector('.to-do-list');
+  }
+
+  static render({ index, description, completed }) {
+    const component = document.createElement('li');
+
+    component.id = index;
+    component.className = 'row';
+
+    if (completed) component.classList.add('status-completed');
+
+    component.innerHTML = `
+      <input type='checkbox'${completed ? ' checked' : ''} class='toggler'>
+      <input type='text' class='field-desc' value='${description}' disabled>
+      <button type='button' class='rm-btn fa-solid fa-ellipsis-vertical'></button>
+    `;
+
+    return component;
+  }
+
+  create(description) {
+    const obj = {
+      index: this.tasks.length,
+      description,
+      completed: false,
+    };
+
+    this.tasks.push(obj);
+    this.listView.appendChild(this.render(obj));
+
+    this.commitTasks();
```

Comment:
- [ ] At the moment, a user is not able to add todo tasks. Kindly ensure that the function to add tasks works as it should.
- [ ] Please also ensure that the functionality for editing and deleting works as it should.